styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -208,14 +218,26 @@ private WorkflowConfiguration readFromJsonWithDefaults(ByteString payload)
     return WorkflowConfigurationBuilder.from(workflowConfig).deploymentTime(time.get()).build();
   }
 
-  private Response<Collection<Workflow>> workflows() {
+  private Response<Collection<Workflow>> workflows(Request request) {
     try {
-      return Response.forPayload(storage.workflows().values());
+      Map<QueryParams, String> paramFilters =
+          Map.of(DEPLOYMENT_TYPE, getFilterParams(request, DEPLOYMENT_TYPE),
+              DEPLOYMENT_TIME_BEFORE, getFilterParams(request, DEPLOYMENT_TIME_BEFORE),
+              DEPLOYMENT_TIME_AFTER, getFilterParams(request, DEPLOYMENT_TIME_AFTER));
+
+      Collection<Workflow> workflows = storage.workflows().values();
+
+      return Response.forPayload(filterWorkflows(workflows, paramFilters));
+
     } catch (IOException e) {
       throw new RuntimeException("Failed to get workflows", e);
     }
   }
 
+  private String getFilterParams(Request request, QueryParams filter) {
+    return request.parameter(filter.getQueryName()).orElse("");
+  }
```

Comment:
```suggestion
  private Optional<String> getFilterParams(Request request, QueryParams filter) {
    return request.parameter(filter.getQueryName());
  }
```
To make it work with my previous comment, maybe now is simple enough that it could be inlined, but I will leave that decision to you